docs/ContributingGuidelines.md

```diff
@@ -0,0 +1,13 @@
+# Contributing Guidelines for Mekanism #
```

Comment:
'i want to fix/add things' should be a completely separate section from 'i have an issue'

And if we're going to have this file it needs to cover *why* they should follow the instructions in the issue template.

Reply:
I agree, and as I'm starting to work on separating the two., I realize that this is a bigger project.  

What I don't know is how to manage a document that is in its infancy.

In my own docs I take a very conversational tone about the current state of a document and where we want to take it in the future.  This works for some people because it feels like a conversation, other people hate it because they just want the information, and they don't want a conversation.

The biggest thing is I don't know what to use for a documentation backlog.  I don't think the issue log is a good place for this "documentation backlog" but I don't know what is the right place.

Branch updated with document changes.